base_local_planner/src/map_grid.cpp

```diff
@@ -285,12 +286,24 @@ namespace base_local_planner{
     }
 
     computeTargetDistance(path_dist_queue, costmap);
-    return mbf_msgs::ExePathResult::SUCCESS;
-  }
 
+    // check if the goal is reachable from the start position
+    if (current_pose) {
+      unsigned int cell_x, cell_y;
+      if ( ! costmap.worldToMap(current_pose->pose.position.x, current_pose->pose.position.y, cell_x, cell_y)) {
+        ROS_ERROR("setLocalGoal: start position out of map");
+        return mbf_msgs::ExePathResult::OUT_OF_MAP;
+      }
 
+      if (this->operator()(cell_x, cell_y).target_dist == unreachableCellCosts()) {
```

Comment:
Q/F: wait, we are not checking here whether the goal is blocked, right?
only current pose

Reply:
So this code has 2 parts:
1) convert the path to a single goal
2) compute distances from the goal (`computeTargetDistance`)

whether the goal is in an obstacle is part of 1), and is checked on line [274](https://github.com/rapyuta-robotics/ros-navigation/blob/bc9ff543d8a487a84af3e2cf10402a16ad117566/base_local_planner/src/map_grid.cpp#L274-L277).

whether the goal is reachable from the start can be done only after 2).